### ISSUE TITLE
Hidden range in nn.Embedding

### ISSUE THREAD

**Issue by PhyllisJi** [pytorch/pytorch#103424]
### Describe the bug

`Hidden range of padding parameter in nn.Embedding`

### Code

```py
import torch
import torch.nn as nn


class BiLSTM(nn.Module):
    def __init__(self, batch_size, hidden_dim, vocab_size, sequence_len):
        super().__init__()

        self.batch_size = batch_size
        self.hidden_dim = hidden_dim
        self.input_size = vocab_size
        self.num_classes = vocab_size
        self.sequence_len = sequence_len

        # Dropout
        self.dropout = nn.Dropout(0.25)

        # Embedding layer
        self.embedding_0 = nn.Embedding(num_embeddings=2, embedding_dim=2, padding_idx=3)

    def forward(self, x):

        # Bi-LSTM
        # hs = [batch_size x hidden_size]
        # cs = [batch_size x hidden_size]
        hs_forward = torch.zeros(x.size(0), self.hidden_dim)
        cs_forward = torch.zeros(x.size(0), self.hidden_dim)
        hs_backward = torch.zeros(x.size(0), self.hidden_dim)
        cs_backward = torch.zeros(x.size(0), self.hidden_dim)

        # LSTM
        # hs = [batch_size x (hidden_size * 2)]
        # cs = [batch_size x (hidden_size * 2)]
        hs_lstm = torch.zeros(x.size(0), self.hidden_dim * 2)
        cs_lstm = torch.zeros(x.size(0), self.hidden_dim * 2)

        # Weights initialization
        torch.nn.init.kaiming_normal_(hs_forward)
        torch.nn.init.kaiming_normal_(cs_forward)
        torch.nn.init.kaiming_normal_(hs_backward)
        torch.nn.init.kaiming_normal_(cs_backward)
        torch.nn.init.kaiming_normal_(hs_lstm)
        torch.nn.init.kaiming_normal_(cs_lstm)

        # From idx to embedding
        x = self.embedding_0(x.long())
        return x


if __name__ == '__main__':
    net = BiLSTM(5, 2, 100, 100)
```

### Version

```
PyTorch version: 2.0.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.10 (main, Mar 21 2023, 13:41:05) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.0.0
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.15.1
[conda] numpy                     1.23.5          py310hb93e574_0  
[conda] numpy-base                1.23.5          py310haf87e8b_0  
[conda] torch                     2.0.0                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.15.1                   pypi_0    pypi
```

**Comment by malfet** [comment]
@PhyllisJi can you please elaborate what do you mean?

**Comment by PhyllisJi** [comment]
> @PhyllisJi can you please elaborate what do you mean?

Yes. I mean that the keyword _padding_idx_ in _nn.Embedding_ is subject to the keyword _num_embedding_ when we are actually programming, but there is no description of this dependency in the documentation.
